src/Main.elm

```diff
@@ -1,10 +1,571 @@
+            { icon = FA.icon FA.folder
+            , isDark = False
+            , cursorPointer = ( "cursor-pointer", True )
+            }
+
+        fileViewConfig : ViewConfig
+        fileViewConfig =
+            { icon = text ""
+            , isDark = True
+            , cursorPointer = ( "cursor-default", True )
+            }
+
+        attachOnClick : List (Attribute Msg)
+        attachOnClick =
+            case viewType of
+                ViewFolder ->
+                    [ onClick toMsg ]
+
+                ViewFile ->
+                    []
+    in
+    div
+        [ class "h-12 bg-gray-100 border-b border-gray-200 hover:bg-gray-600"
+        , classList
+            [ ( "bg-gray-200", config.isDark )
+            , ( "border-gray-300", config.isDark )
+            , config.cursorPointer
+            ]
+        ]
+        [ div ([ class "group w-5/6 h-full mx-auto flex items-center justify-between" ] ++ attachOnClick)
+            [ span [ class "font-semibold text-gray-800 leading-loose group-hover:text-white" ] [ text name ]
+            , span [ class "text-gray-400 group-hover:text-white" ] [ config.icon ]
+            ]
+        ]
+
+
+foldersOnly : Node -> Bool
+foldersOnly node =
+    case node of
+        Dir id audienceFolder ->
+            True
+
+        File id audience ->
+            False
+
+        Root ->
+            True
+
+
+filesOnly : Node -> Bool
+filesOnly node =
+    case node of
+        Dir id audienceFolder ->
+            False
+
+        File id audience ->
+            True
+
+        Root ->
+            True
+
+
+
+{------------------------------
+------------------------------
+DECODE JSON 
+
+Data Type : 
+Data.Audience.Audience
+Data.AudienceFolder.AudienceFolder
+
+JSON strings fetched from
+Data.Audience.audiencesJSON
+Data.AudienceFolder.audienceFoldersJSON
+
+Decoding JSON into two formats List Audience and List AudienceFolder 
+
+------------------------------
+------------------------------}
+-- Decode Audience
+
+
+decodeAudiences : Maybe (List Audience)
+decodeAudiences =
+    case decodeString audiencesDecoder audiencesJSON of
+        Ok result ->
+            Just result
+
+        Err _ ->
+            Nothing
+
+
+audiencesDecoder : Decoder (List Audience)
+audiencesDecoder =
+    Decode.succeed Audience
+        |> required "id" int
+        |> required "name" string
+        |> required "type" audienceTypeDecoder
+        |> required "folder" (nullable int)
+        |> list
+        |> field "data"
+
+
+audienceTypeDecoder : Decoder AudienceType
+audienceTypeDecoder =
+    string
+        |> Decode.andThen strToAudienceType
+
+
+strToAudienceType : String -> Decoder AudienceType
+strToAudienceType name =
+    case name of
+        "curated" ->
+            Decode.succeed Curated
+
+        "shared" ->
+            Decode.succeed Shared
+
+        "user" ->
+            Decode.succeed Authored
+
+        _ ->
+            Decode.fail "\"type\" is malformed"
```

Comment:
Nice. As a bonus, it's sometimes useful to show what the failing value was in `Decode.fail`:

```elm
Decode.fail <| "\"type\" is malformed: " ++ name
```
```